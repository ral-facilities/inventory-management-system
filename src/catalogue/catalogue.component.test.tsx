import React from 'react';
import {
  renderComponentWithBrowserRouter,
  renderComponentWithMemoryRouter,
} from '../setupTests';
import { screen, waitFor } from '@testing-library/react';
import Catalogue from './catalogue.component';
import userEvent from '@testing-library/user-event';

describe('Catalogue', () => {
  let user;
  const createView = () => {
    return renderComponentWithBrowserRouter(<Catalogue />);
  };

  beforeEach(() => {
    user = userEvent.setup();
  });

  it('renders text correctly', () => {
    createView();
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });
  it('navigates back to the root directory', async () => {
    createView();

    const homeButton = await screen.findByTestId('home-button-catalogue');
    await user.click(homeButton);
    expect(window.location.pathname).toBe('/catalogue');
  });
  it('opens the add catalogue category dialog', async () => {
    createView();

    const addButton = await screen.findByTestId('add-button-catalogue');
    await user.click(addButton);

    await waitFor(() => {
      expect(screen.getByRole('dialog')).toBeInTheDocument();
    });

    const closeButton = screen.getByRole('button', { name: 'Cancel' });
    await user.click(closeButton);
    await waitFor(() => {
      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    });
  });
  it('renders the breadcumbs and navigate to another directory', async () => {
    renderComponentWithMemoryRouter(
      <Catalogue />,
      '/catalogue/motion/actuators'
    );

    await waitFor(() => {
      expect(screen.getByRole('link', { name: 'motion' })).toBeInTheDocument();
    });
    await user.click(screen.getByRole('link', { name: 'motion' }));

    await waitFor(() => {
      expect(
        screen.queryByRole('link', { name: 'motion' })
      ).not.toBeInTheDocument();
    });
  });
});
