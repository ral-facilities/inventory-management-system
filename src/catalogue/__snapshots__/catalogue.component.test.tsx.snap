// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalogue renders text correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1902l77"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wtqo72-MuiButtonBase-root-MuiButton-root"
      data-testid="home-button-catalogue"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="HomeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiBox-root css-1d92c1"
    >
      <div
        class="MuiBox-root css-0"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-h1b7t3-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
              />
            </li>
          </ol>
        </nav>
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1nq2t50-MuiSvgIcon-root"
      data-testid="NavigateNextIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wtqo72-MuiButtonBase-root-MuiButton-root"
      data-testid="add-button-catalogue"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
